src/ig/IGExporter.ts

```diff
@@ -155,6 +155,18 @@ export class IGExporter {
         exampleBoolean: false
       });
     });
+    this.pkg.instances.forEach(instance => {
+      const instanceFileName = `${instance.resourceType}-${instance.id ?? instance.instanceName}`;
```

Comment:
I think you could make a case here that the `instance.instanceName` should always be used to name the file. If you check out something like USCore, https://www.hl7.org/fhir/medicationadministration0311.json.html, you will see that the file is not necessarily named after the id of the instance. Also I don't think there would necessarily be any rule in FHIR against having two instances with the same id, which could cause an issue here (although I think the example of USCore is a stronger reason to always use instanceName).

Reply:
Those are two good points. I think when I originally discussed the naming pattern with @cmoesel, we were looking at the USCore patient example [here](https://www.hl7.org/fhir/us/core/Patient-example.json.html) and followed that. Given that doesn't seem consistent, I don't think I have a strong opinion either way. Are the `instanceName`s given in the FSH file supposed to be unique? Because if so, that seems like a fair enough reason to just use that.

Reply:
I talked more with @ngfreiter about this, and I think we are going to leave the naming as it is for now. Using the specified `id` if it exists seems to mirror what USCore and mCODE are using as file names for various instances.